.github/workflows/test.yml

```diff
@@ -2,35 +2,46 @@ name: Test
 
 on:
   pull_request:
-    paths:
-      - src/**
-      - tests/**
-      - 'phpunit.xml*'
-      - 'composer.*'
+    branches:
+      - 'master'
+    paths-ignore:
+      - '**.md'
   workflow_dispatch:
 
+concurrency:
+  group: '${{ github.workflow }}-${{ github.ref }}'
+  cancel-in-progress: true
+
 jobs:
   integration:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        php-version: ['5.6', '7.4', '8.0', '8.1']
+        php-version: ['5.6', '7.4', '8.0', '8.1', '8.2', '8.3']
+
+    name: Unit tests on PHP ${{ matrix.php-version }}
+
+    continue-on-error: ${{ matrix.php-version == '8.3' }}
+
     steps:
       - uses: actions/checkout@v3
-      
+
       - uses: shivammathur/setup-php@v2
         with:
-          tools: composer
           php-version: ${{ matrix.php-version }}
 
+      - name: Install Composer dependencies
+        uses: ramsey/composer-install@v2
+        with:
+          # Bust the cache at least once a month - output format: YYYY-MM.
+          custom-cache-suffix: $(date -u "+%Y-%m")
```

Comment:
Can you explain why we'd want to do this? I would expect the cache to be based on the dependencies that were installed which shouldn't change over time (for the same versions of course).

Reply:
I think this comment explains it far better than I could: https://github.com/PHPMailer/PHPMailer/pull/2821 😄

Reply:
Thanks, that's really useful context 👍